torch/csrc/autograd/profiler.h

```diff
@@ -24,16 +29,93 @@ constexpr inline std::size_t ceilToMultiple(std::size_t a, std::size_t b) {
   return ((a + b - 1) / b) * b;
 }
 
+inline uint64_t getTime() {
+  using namespace std::chrono;
+  using clock = std::conditional<high_resolution_clock::is_steady, high_resolution_clock, steady_clock>::type;
+  return duration_cast<nanoseconds>(clock::now().time_since_epoch()).count();
+}
+
 enum class EventKind {
   Mark,
   PushRange,
   PopRange
 };
+#ifdef WITH_CUDA
+static void cudaCheck(cudaError_t result, const char * file, int line) {
+  if(result != cudaSuccess) {
+    std::stringstream ss;
+    ss << file << ":" << line << ": " << cudaGetErrorString(result);
+    throw std::runtime_error(ss.str());
+  }
+}
+#define PROFILER_CUDA_CHECK(result) cudaCheck(result,__FILE__,__LINE__);
+#endif
 
-// NOTE: we don't need a flag saying if an event is a kernel, because it's
-// used only for the CPU-side perf recording.
-using Event = std::tuple<std::string, uint64_t, EventKind>; // (name, time, kind)
+struct Event {
+  Event(EventKind kind, std::string name, uint32_t thread_id, bool record_cuda)
+  : kind_(kind),
+  name_(std::move(name)),
+  thread_id_(thread_id),
+  cpu_ns(getTime()) {
```

Comment:
can you please put the colon at the end of the line like you do with commas, or put all the commas at the beginning?

Reply:
Also, can you please use trailing underscores either for all or no members?